crates/subspace-node/src/bin/subspace-node.rs

```diff
@@ -421,6 +421,7 @@ fn main() -> Result<(), Error> {
 
                         DsnConfig {
                             keypair,
+                            base_path: cli.run.shared_params.base_path.clone(),
```

Comment:
This is not the right way to use it, you should call `cli.run.base_path()` instead that will also support `--tmp` and `--dev` flags, there are a few examples of this already.

It'll also be potentially shared across different chains concurrently, so likely needs to be nested like this (otherwise at very least wiping needs to be updated to clean this up): https://github.com/paritytech/substrate/blob/master/client/cli/src/config.rs#L490

Reply:
I'm not sure I understood this and the next comment 100%. Please, review my changes.